sdv/metadata/multi_table.py

```diff
@@ -333,6 +368,87 @@ def _log_detected_table(single_table_metadata):
         table_json = json.dumps(table_dict, indent=4)
         LOGGER.info(f'Detected metadata:\n{table_json}')
 
+    def _validate_all_tables_connected(self, parent_map, child_map):
+        """Get the connection status of all tables.
+
+        Args:
+            parent_map (dict):
+                Dictionary mapping each parent table to its child tables.
+            child_map (dict):
+                Dictionary mapping each child table to its parent tables.
+
+        Returns:
+            dict specifying whether each table is connected the other tables.
+        """
+        nodes = list(self.tables.keys())
+        if len(nodes) == 1:
+            return
+
+        queue = [nodes[0]]
+        explored = set()
+        connected = {table_name: False for table_name in nodes}
+
+        while queue:
+            node = queue.pop()
+            explored.add(node)
+            children = list(child_map[node]) + list(parent_map[node])
+            if children:
+                connected[node] = True
+                for child in children:
+                    if not connected[child] and child not in queue:
+                        queue.append(child)
+            elif not queue:  # This was an orphan node, pick a new eligible node
+                for eligible_node in nodes:
+                    if eligible_node not in explored:
+                        queue.append(eligible_node)
+                        break
```

Comment:
I think we could achieve this if we picked are starting node from the parent map right? Then it's guaranteed to be valid

Reply:
Yeah that makes sense, it also makes it easier to handle the situation where no tables are connected.

Reply:
Actually after looking again, I'm wondering if it's a problem that we start on a node with no neighbors. If that happens then the check below will still fail since the rest of the tables aren't connected to the first one right?

Reply:
It does fail, but I think the error message is confusing. If we have just 1 orphan table and we pick that as the starting node, the error message we get says that every other table is not connected to any of the other tables. If we don't have any relationships, I think it makes sense to list every table as not connected to any of the others.